### PR TITLE
Refactor Quantization Aware Training of TF backend

### PR DESCRIPTION
**Pull request by zehao-intel**
Signed-off-by: zehao-intel <zehao.huang@intel.com>

## Type of Change

feature or bug fix or documentation or validation or others  
API changed or not: Yes

## Description

detail description 
JIRA ticket: 2304

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR : as titled

## How has this PR been tested?

how to reproduce the test (including hardware information): extension

## Dependency Change?

any library dependency introduced or removed: yes, tfmot removed

### CODE REVIEW

**Comment by zehao-intel** [comment]
Extension: https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/3688/

File: examples/tensorflow/image_recognition/keras_models/mnist/quantization/qat/main.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        compression_manager.save(FLAGS.output_model)
+
+    if FLAGS.benchmark:
+        from neural_compressor.benchmark import fit
+        from neural_compressor.experimental import common
+        from neural_compressor.config import BenchmarkConfig
+        assert FLAGS.mode == 'performance' or FLAGS.mode == 'accuracy', \
+        "Benchmark only supports performance or accuracy mode."
+
+        model = common.Model(FLAGS.input_model).graph_def
```
Comment (lvliang-intel) [inline examples/tensorflow/image_recognition/keras_models/mnist/quantization/qat/main.py:169]: common.Model needs to be updated after https://github.com/intel/neural-compressor/pull/244 merged.

File: examples/tensorflow/image_recognition/keras_models/resnet50/quantization/qat/README [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                        epochs=1)
+
+    # apply some post process steps and save the output model
+    compression_manager.callbacks.on_train_end()
+    compression_manager.save(FLAGS.output_model)
+```
+#### Benchmark
+```python
+    from neural_compressor.benchmark import fit
+    from neural_compressor.experimental import common
```
Comment (lvliang-intel) [inline examples/tensorflow/image_recognition/keras_models/resnet50/quantization/qat/README:105]: from neural_compressor.experimental import common, 
this needs to be updated after https://github.com/intel/neural-compressor/pull/244 merged.

File: examples/tensorflow/image_recognition/keras_models/resnet50/quantization/qat/main.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+flags.DEFINE_bool(
+    'benchmark', False, 'whether to benchmark the model')
+
+flags.DEFINE_string(
+    'dataset_location', None, 'location of the dataset on tfrecord format')
+
+flags.DEFINE_integer(
+    'batch_size', 32, 'batch_size')
+
+from neural_compressor.experimental.metric.metric import TensorflowTopK
```
Comment (lvliang-intel) [inline examples/tensorflow/image_recognition/keras_models/resnet50/quantization/qat/main.py:52]: experimental all needs to be updated after https://github.com/intel/neural-compressor/pull/244 merged.

Comment (zehao-intel) [reply examples/tensorflow/image_recognition/keras_models/resnet50/quantization/qat/main.py:52]: They will be modified if PR244 can be merged before code freeze.

**Comment by zehao-intel** [comment]
Extension: https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/3694/

**Pull request merged**